api/jobs_test.go

```diff
@@ -1219,19 +1209,108 @@ func TestPutNumTasksHandler(t *testing.T) {
 			UnlockJobFunc: func(ctx context.Context, lockID string) {
 				// mock UnlockJob to be successful by doing nothing
 			},
+			GetJobFunc: func(ctx context.Context, id string) (*models.Job, error) {
+				switch id {
+				case notFoundJobID:
+					return nil, mongo.ErrJobNotFound
+				default:
+					jobs := expectedJob(ctx, t, cfg, false, id, "", 0)
+					return &jobs, nil
+				}
+			},
+			UpdateJobFunc: func(ctx context.Context, id string, updates bson.M) error {
+				switch id {
+				case validJobID2:
+					return nil
+				default:
+					return errUnexpected
+				}
+			},
 		}
 
 		httpClient := dpHTTP.NewClient()
 		apiInstance := api.Setup(mux.NewRouter(), jobStoreMock, &apiMock.AuthHandlerMock{}, taskNames, cfg, httpClient, &apiMock.IndexerMock{}, &apiMock.ReindexRequestedProducerMock{})
 
-		Convey("When a request is made to update the number of tasks of a specific job that exists in the Data Store", func() {
+		Convey("When a request is made to update the number of tasks of a specific job and with no If-Match header set", func() {
+			req := httptest.NewRequest("PUT", fmt.Sprintf("http://localhost:25700/jobs/%s/number_of_tasks/%s", validJobID2, validCount), nil)
+			resp := httptest.NewRecorder()
+
+			apiInstance.Router.ServeHTTP(resp, req)
+
+			Convey("Then a status code 200 is returned", func() {
+				So(resp.Code, ShouldEqual, http.StatusOK)
+
+				Convey("And the etag of the response job resource should be returned via the ETag header", func() {
+					So(resp.Header().Get(dpresponse.ETagHeader), ShouldNotBeEmpty)
+				})
+			})
+		})
+
+		Convey("When a request is made where If-Match header is set to *", func() {
 			req := httptest.NewRequest("PUT", fmt.Sprintf("http://localhost:25700/jobs/%s/number_of_tasks/%s", validJobID2, validCount), nil)
+			headers.SetIfMatch(req, "*")
```

Comment:
Please handle SetIfMatch error here (line 1251) and also for lines 1268, 1285, and 1301.

Reply:
Done